### PR TITLE
[azure-docs] refine AzureBlobComputeLogManager guide

### PR DESCRIPTION
**Pull request by mlarose**
## Summary & Motivation

Updates the Azure Blob Storage compute logs documentation to clarify the setup process and fix issues with workload identity configuration. Adds specific instructions for using the Dagster+ Helm chart and includes a new "Next steps" section in the ACR user code guide.

## How I Tested These Changes

- Verified all commands and configurations work in an Azure Kubernetes Service (AKS) environment

## Changelog

- NOCHANGELOG

### CODE REVIEW

**Comment by mlarose** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26407?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26407** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26407?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26407?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26397** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26397?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 1]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
-  --subject system:serviceaccount:dagster-agent:dagster-agent-service-account
+  --subject system:serviceaccount:<dagster-agent-namespace>:<dagster-agent-service-account>
 ```
 
-Finally, we'll edit our AKS agent deployment to use the new service account.
+You will need to obtain the client id of this identity for the next few operations:
 
 ```bash
-kubectl edit deployment <your-user-cloud-deployment> -n dagster-agent
+az identity show -g <resource-group> -n agent-identity --query 'clientId' -otsv
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:54]: maybe a note to save this value for later.

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 2]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
 ```
 
-In the deployment manifest, add the following lines:
+We need to grant access to the storage account. Replace `<storage-account>` with the name of your storage account.
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:57]: Need to replace a few different values here, not just storage-account

Comment (dpeng817) [reply docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:57]: Not sure this particular sentence is necessary, I think the example is self explanatory

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 3]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
-    ...
-    spec:
-      ...
-      serviceAccountName: dagster-agent-sa
 ```
 
-If everything is set up correctly, you should be able to run the following command and see an access token returned:
+<Note>
+  If you need to retrieve the current values used by your Helm deployment, you
+  can use this command: `helm get values user-cloud > current-values.yaml`.
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:84]: maybe make a note that this will pull down the values into a file current-values.yaml

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 4]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
+</Note>
+
+Finally, update your helm release with the new values:
 
 ```bash
-kubectl exec -n dagster-agent -it <pod-in-cluster> -- bash
-# in the pod
-curl -H "Metadata:true" "http://169.254.169.254/metadata/identity/oauth2/token?resource=https://storage.azure.com/"
+helm upgrade user-cloud dagster-cloud/dagster-cloud-agent -n <dagster-agent-namespace> -f values.yaml
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:90]: should probably also use current-values.yaml here

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 5]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
-Now, you need to update the helm values to use Azure Blob Storage for logs. You can do this by editing the `values.yaml` file for your user-cloud deployment.
-
-Pull down the current values for your deployment:
+If everything is set up correctly, you should be able to run the following command and see an access token returned:
 
 ```bash
-helm get values user-cloud > current-values.yaml
+kubectl exec -n <dagster-agent-namespace> -it <pod-in-cluster> -- bash
+# in the pod
+apt update && apt install -y curl
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:98]: is there a chance this won't work for some users? Might be worth caveating that installation of curl might be different depending on distro

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 6]
```diff
@@ -97,18 +114,22 @@ computeLogs:
...
 
 ```bash
-helm upgrade user-cloud dagster-cloud/dagster-cloud-agent -n dagster-agent -f current-values.yaml
+helm upgrade user-cloud dagster-cloud/dagster-cloud-agent -n <dagster-agent-namespace> -f values.yaml
 ```
 
-## Step 3: Verify logs are being written to Azure Blob Storage
+## Step 3: Update your code location to enable the use of the AzureBlobComputeLogManager
+
+- Add `dagster-azure` to your `setup.py` file. This will allow you to import the `AzureBlobComputeLogManager` class.
```
Comment (dpeng817) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:130]: Good note.

**Review by dpeng817 — APPROVED** [review]
Awesome stuff, man. I think this is all super valuable and thanks again for dogfooding.

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 7]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
   --name dagster-agent-federated-id \
   --identity-name agent-identity \
   --resource-group <resource-group> \
   --issuer $(az aks show -g <resource-group> -n <aks-cluster-name> --query "oidcIssuerProfile.issuerUrl" -otsv) \
-  --subject system:serviceaccount:dagster-agent:dagster-agent-service-account
+  --subject system:serviceaccount:<dagster-agent-namespace>:<dagster-agent-service-account>
 ```
 
-Finally, we'll edit our AKS agent deployment to use the new service account.
+You will need to obtain the client id of this identity for the next few operations:
```
Comment (neverett) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:51]:
```suggestion
You will need to obtain the client id of this identity for the next few operations. Make sure to save this value:
```

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx (hunk 2 above)
Comment (neverett) [reply docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:57]: Yeah, I think you can leave this sentence out in this particular guide, given the likely audience.

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 8]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
+We need to grant access to the storage account. Replace `<storage-account>` with the name of your storage account.
+
+```bash
+az role assignment create \
+  --assignee <managed-identity-client-id> \
+  --role "Storage Blob Data Contributor" \
+  --scope $(az storage account show -g <resource-group> -n <storage-account> --query 'id' -otsv)
+```
+
+New annotations and labels need to be added in Kubernetes to enable the use of workload identity. If you're using the Dagster+ Helm Chart, you need to modify your values.yaml to add the following lines:
```
Comment (neverett) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:66]: I'd rephrase this in the active voice: "You will need to add new annotations and labels in Kubernetes to enable the use of workload identities"

Comment (neverett) [reply docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:66]: Second sentence could be tightened up a bit by removing "you need to"

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx (hunk 3 above)
Comment (neverett) [reply docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:84]: I would replace "use this command" with "run"

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 9]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
-      serviceAccountName: dagster-agent-sa
 ```
 
-If everything is set up correctly, you should be able to run the following command and see an access token returned:
+<Note>
+  If you need to retrieve the current values used by your Helm deployment, you
+  can use this command: `helm get values user-cloud > current-values.yaml`.
+</Note>
+
+Finally, update your helm release with the new values:
```
Comment (neverett) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:87]: Capitalize Helm for consistency

File: docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx [hunk 10]
```diff
@@ -25,66 +25,83 @@ First, we'll enable the cluster to use workload identity. This will allow the AK
...
+kubectl exec -n <dagster-agent-namespace> -it <pod-in-cluster> -- bash
+# in the pod
+apt update && apt install -y curl
+curl -H "Metadata:true" "http://169.254.169.254/metadata/identity/oauth2/token?resource=https://storage.azure.com/&api-version=2018-02-01"
 ```
 
-Then, edit the `current-values.yaml` file to include the following lines:
+## Step 2: Configure Dagster to use Azure Blob Storage
+
+Once again, you need to update the helm values to use Azure Blob Storage for logs. You can do this by editing the `values.yaml` file for your user-cloud deployment to include the following lines:
```
Comment (neverett) [inline docs/content/dagster-plus/deployment/azure/blob-compute-logs.mdx:104]: Capitalize Helm

**Review by neverett — APPROVED** [review]
I left a few comments that you can apply as you see fit, but overall this looks good to me. I'll go ahead and approve so you're not blocked on merging this.

**Comment by mlarose** [comment]
Thanks Nikki, I appreciate the feedback!

**Comment by mlarose** [comment]
### Merge activity

* **Dec 12, 2:10 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26407).
